### PR TITLE
Relax dependency pinning

### PR DESCRIPTION
**Pull request by dhirschfeld**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [ ] adds new tests (if appropriate)
- [ ] updates `CHANGELOG.md` (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR relaxes some if the max pins on the dependencies

## Why is this PR important?
Currently installing `prefect` into an existing environment requires downgrading a number of dependencies.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
I'm not sure if there was a good reason for the specific max-pins so I thought I'd get the CI to check that it works with the latest versions...

**Comment by dhirschfeld** [comment]
Not sure what happened there - CircleCI seemed to hang after running the tests:
```
= 3654 passed, 13 skipped, 1 xfailed, 30 xpassed, 4 warnings in 61.32 seconds ==
Too long with no output (exceeded 10m0s)
```

Will re-open to try again...

**Comment by dhirschfeld** [comment]
Not sure why CircleCI is hanging but the tests seem to be passing with the updated deps:
```
= 3654 passed, 13 skipped, 1 xfailed, 30 xpassed, 4 warnings in 61.32 seconds ==
```

**Comment by cicdw** [comment]
Every so often, the python 3.7 job hangs like that.  We were able to reduce the frequency by removing code coverage from `pytest` in that particular job, but it still occurs every ~5-10 runs.

File: requirements.txt [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 dask >= 0.18, < 1.2.2
 distributed >= 1.26.1, < 2.0
-docker >= 3.4.1, < 4.0
-idna < 2.8, >= 2.5
+docker >= 3.4.1, < 4.1
+idna >= 2.5, < 2.9
 marshmallow == 3.0.0b19
 marshmallow-oneofschema >= 2.0.0b2, < 3.0
-mypy >= 0.600, < 0.700
+mypy >= 0.600, < 0.800
```
Comment (cicdw) [inline requirements.txt]: You're welcome to test out removing this for #1065

**Comment by cicdw** [comment]
Awesome, thanks @dhirschfeld ! 

I'm going to have to test this against some other systems tomorrow, but what I can say is that the reason `docker` and `pyyaml` are capped where they are was _originally_ due to [`docker-compose`'s requirements](https://github.com/docker/compose/blob/master/requirements.txt).

**Comment by dhirschfeld** [comment]
Wow, that's a *terrible* `requirements.txt`! Exact pinning is a sure-fire way to break an environment - at least for `conda` where consistency is enforced.

That's necessarily relaxed for the conda package:
https://github.com/conda-forge/docker-compose-feedstock/blob/master/recipe/meta.yaml

**Comment by jlowin** [comment]
@dhirschfeld we have a support group at Prefect for headaches caused by docker-compose pins... you are very welcome to join :)

**Comment by dhirschfeld** [comment]
The main one I care about is `cloudpickle` nevertheless, I'd like to better understand the issue:
> *Installing `prefect` (with `pip`) before a package with a `docker-compose` requirement leads to a corrupted environment*

What is the actual problem here? Since the `docker-compose` has no version pinning at all with the `conda` package it would suggest that `docker-compose` can run fine with that latest versions of all dependencies.

If that wasn't the case then you'd expect `conda`'s behaviour of installing the latest possible versions would break the `docker-compose` `conda` package.

If `docker-compose` can actually run fine with the latest dependencies is the problem that mis-specified dependencies in the `docker-compose` [`requirements.txt`](https://github.com/docker/compose/blob/master/requirements.txt) breaks `pip`

i.e. a post-prefect install of `docker-compose` will enforce the exact pins in the `requirements.txt` which are incompatible with `prefect`?

**Comment by cicdw** [comment]
Good question - here's my quick response that might require a reproducible example to fully justify: I agree that if a user runs `pip install prefect` and then `pip install docker-compose`, the dependencies will be sane and consistent (prefect shouldn't conflict with the pins in `docker-compose`).  _However_, because `pip` doesn't actually do dependency resolution, if I instead run `pip install prefect` and then `pip install MY_OTHER_PACKAGE`, and `MY_OTHER_PACKAGE` depends on `docker-compose`, the environment will now have some conflicts.  And we've seen in the past that certain CLI entry points could break (with a `pkg_resources.DistributionNotFound` error) if the environment doesn't contain the correct package requirements, so that's the concrete consequence I'm most nervous about.

**Comment by dhirschfeld** [comment]
Maybe the way forward for now is to hold off on relaxing the pip requirements and only make the changes in the [`conda`](https://github.com/conda-forge/prefect-feedstock/blob/master/recipe/meta.yaml) package?

**Comment by cicdw** [comment]
Yea that sounds good - we can definitely relax `mypy` and `cloudpickle` still, too

File: requirements.txt (hunk 1 above)
Comment (dhirschfeld) [reply requirements.txt]: Once this is merged I'll give that a go...

**Comment by cicdw** [comment]
@dhirschfeld would you like to add your name to the Contributors section of the CHANGELOG for this release?  I'll leave this open for a bit in case you'd like to.

Also, thanks a lot for the PR!

**Comment by dhirschfeld** [comment]
> *would you like to add your name to the Contributors section of the CHANGELOG*

Thanks - have gone ahead and done so!

**Pull request merged**